homeassistant/components/bluetooth/base_scanner.py

```diff
@@ -103,7 +103,8 @@ def _async_watchdog_triggered(self) -> bool:
         )
         return time_since_last_detection > SCANNER_WATCHDOG_TIMEOUT
 
-    async def _async_scanner_watchdog(self, now: datetime.datetime) -> None:
+    @hass_callback
+    def _async_scanner_watchdog(self, now: datetime.datetime) -> None:
```

Comment:
Potentially breaking for custom components?

Reply:
Its not a breaking change if merged before 2023.1.0